datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/PanelRecord.kt

```diff
@@ -16,7 +17,7 @@ data class PanelRecord(
     override val hasSufficientQuality: Boolean
 ) : MolecularTest {
 
-    override fun testsGene(gene: String): Boolean {
-        return testedGenes.contains(gene)
+    override fun testsGene(gene: String, molecularTestTargets: Predicate<List<MolecularTestTarget>>): Boolean {
+        return geneSpecifications[gene]?.let { molecularTestTargets.test(it) } ?: false
```

Comment:
Although does look a little easier, would sleep better if this would be unit-tested :)